src/service/ToolService.ts

```
@@ -21,241 +23,379 @@
+        console.error('Error fetching YouTube transcript:', error);
+				return null;
+			}
+		},
+	};
+	
+	/**
+	 * Maps tool names to their definitions, including Zod schema, description, and function.
+	 */
+	// deno-lint-ignore ban-types
+	static tools = new Map<string, { description: string; zodSchema: ZodObject<any, any, any>; fn: Function }>([
+		['search_searx', ToolService.searchSearxTool],
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)